### ISSUE TITLE
Use ConfigData.Option.IGNORE_IMPORTS

### ISSUE THREAD

**Issue by spencergibb** [spring-cloud/spring-cloud-config#1793]
See https://github.com/spring-cloud/spring-cloud-config/issues/1788#issuecomment-763171503

**Comment by spencergibb** [comment]
If I add that I get the following:
```

java.lang.NullPointerException: null
	at org.springframework.boot.context.config.ConfigDataEnvironmentContributor.withBoundProperties(ConfigDataEnvironmentContributor.java:201) ~[spring-boot-2.4.2-20210111.155855-61.jar:2.4.2-SNAPSHOT]
	at org.springframework.boot.context.config.ConfigDataEnvironmentContributors.withProcessedImports(ConfigDataEnvironmentContributors.java:109) ~[spring-boot-2.4.2-20210111.155855-61.jar:2.4.2-SNAPSHOT]
	at org.springframework.boot.context.config.ConfigDataEnvironment.processWithProfiles(ConfigDataEnvironment.java:309) ~[spring-boot-2.4.2-20210111.155855-61.jar:2.4.2-SNAPSHOT]
	at org.springframework.boot.context.config.ConfigDataEnvironment.processAndApply(ConfigDataEnvironment.java:234) ~[spring-boot-2.4.2-20210111.155855-61.jar:2.4.2-SNAPSHOT]
	at org.springframework.boot.context.config.ConfigDataEnvironmentPostProcessor.postProcessEnvironment(ConfigDataEnvironmentPostProcessor.java:97) ~[spring-boot-2.4.2-20210111.155855-61.jar:2.4.2-SNAPSHOT]
	at org.springframework.boot.context.config.ConfigDataEnvironmentPostProcessor.postProcessEnvironment(ConfigDataEnvironmentPostProcessor.java:89) ~[spring-boot-2.4.2-20210111.155855-61.jar:2.4.2-SNAPSHOT]
	at org.springframework.boot.env.EnvironmentPostProcessorApplicationListener.onApplicationEnvironmentPreparedEvent(EnvironmentPostProcessorApplicationListener.java:100) ~[spring-boot-2.4.2-20210111.155855-61.jar:2.4.2-SNAPSHOT]
	at org.springframework.boot.env.EnvironmentPostProcessorApplicationListener.onApplicationEvent(EnvironmentPostProcessorApplicationListener.java:86) ~[spring-boot-2.4.2-20210111.155855-61.jar:2.4.2-SNAPSHOT]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:176) ~[spring-context-5.3.3-20210111.155059-45.jar:5.3.3-SNAPSHOT]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:169) ~[spring-context-5.3.3-20210111.155059-45.jar:5.3.3-SNAPSHOT]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:143) ~[spring-context-5.3.3-20210111.155059-45.jar:5.3.3-SNAPSHOT]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:131) ~[spring-context-5.3.3-20210111.155059-45.jar:5.3.3-SNAPSHOT]
	at org.springframework.boot.context.event.EventPublishingRunListener.environmentPrepared(EventPublishingRunListener.java:82) ~[spring-boot-2.4.2-20210111.155855-61.jar:2.4.2-SNAPSHOT]
	at org.springframework.boot.SpringApplicationRunListeners.lambda$environmentPrepared$2(SpringApplicationRunListeners.java:63) ~[spring-boot-2.4.2-20210111.155855-61.jar:2.4.2-SNAPSHOT]
	at java.util.ArrayList.forEach(ArrayList.java:1257) ~[na:1.8.0_252]
	at org.springframework.boot.SpringApplicationRunListeners.doWithListeners(SpringApplicationRunListeners.java:117) ~[spring-boot-2.4.2-20210111.155855-61.jar:2.4.2-SNAPSHOT]
	at org.springframework.boot.SpringApplicationRunListeners.doWithListeners(SpringApplicationRunListeners.java:111) ~[spring-boot-2.4.2-20210111.155855-61.jar:2.4.2-SNAPSHOT]
	at org.springframework.boot.SpringApplicationRunListeners.environmentPrepared(SpringApplicationRunListeners.java:62) ~[spring-boot-2.4.2-20210111.155855-61.jar:2.4.2-SNAPSHOT]
	at org.springframework.boot.SpringApplication.prepareEnvironment(SpringApplication.java:362) ~[spring-boot-2.4.2-20210111.155855-61.jar:2.4.2-SNAPSHOT]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:320) ~[spring-boot-2.4.2-20210111.155855-61.jar:2.4.2-SNAPSHOT]
	at org.springframework.boot.test.context.SpringBootContextLoader.loadContext(SpringBootContextLoader.java:123) [spring-boot-test-2.4.2-20210111.155855-61.jar:2.4.2-SNAPSHOT]
	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContextInternal(DefaultCacheAwareContextLoaderDelegate.java:99) [spring-test-5.3.3-20210111.155059-45.jar:5.3.3-SNAPSHOT]
	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContext(DefaultCacheAwareContextLoaderDelegate.java:124) [spring-test-5.3.3-20210111.155059-45.jar:5.3.3-SNAPSHOT]
	at org.springframework.test.context.support.DefaultTestContext.getApplicationContext(DefaultTestContext.java:124) [spring-test-5.3.3-20210111.155059-45.jar:5.3.3-SNAPSHOT]
	at org.springframework.test.context.support.DependencyInjectionTestExecutionListener.injectDependencies(DependencyInjectionTestExecutionListener.java:118) [spring-test-5.3.3-20210111.155059-45.jar:5.3.3-SNAPSHOT]
	at org.springframework.test.context.support.DependencyInjectionTestExecutionListener.prepareTestInstance(DependencyInjectionTestExecutionListener.java:83) [spring-test-5.3.3-20210111.155059-45.jar:5.3.3-SNAPSHOT]
	at org.springframework.boot.test.autoconfigure.SpringBootDependencyInjectionTestExecutionListener.prepareTestInstance(SpringBootDependencyInjectionTestExecutionListener.java:43) [spring-boot-test-autoconfigure-2.4.2-20210111.155855-61.jar:2.4.2-SNAPSHOT]
	at org.springframework.test.context.TestContextManager.prepareTestInstance(TestContextManager.java:244) [spring-test-5.3.3-20210111.155059-45.jar:5.3.3-SNAPSHOT]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.createTest(SpringJUnit4ClassRunner.java:227) [spring-test-5.3.3-20210111.155059-45.jar:5.3.3-SNAPSHOT]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner$1.runReflectiveCall(SpringJUnit4ClassRunner.java:289) [spring-test-5.3.3-20210111.155059-45.jar:5.3.3-SNAPSHOT]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit-4.13.1.jar:4.13.1]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.methodBlock(SpringJUnit4ClassRunner.java:291) [spring-test-5.3.3-20210111.155059-45.jar:5.3.3-SNAPSHOT]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:246) [spring-test-5.3.3-20210111.155059-45.jar:5.3.3-SNAPSHOT]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:97) [spring-test-5.3.3-20210111.155059-45.jar:5.3.3-SNAPSHOT]
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331) [junit-4.13.1.jar:4.13.1]
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79) [junit-4.13.1.jar:4.13.1]
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329) [junit-4.13.1.jar:4.13.1]
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66) [junit-4.13.1.jar:4.13.1]
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293) [junit-4.13.1.jar:4.13.1]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26) [junit-4.13.1.jar:4.13.1]
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61) [spring-test-5.3.3-20210111.155059-45.jar:5.3.3-SNAPSHOT]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) [junit-4.13.1.jar:4.13.1]
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70) [spring-test-5.3.3-20210111.155059-45.jar:5.3.3-SNAPSHOT]
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306) [junit-4.13.1.jar:4.13.1]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413) [junit-4.13.1.jar:4.13.1]
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:190) [spring-test-5.3.3-20210111.155059-45.jar:5.3.3-SNAPSHOT]
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137) [junit-4.13.1.jar:4.13.1]
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69) [junit-rt.jar:na]
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33) [junit-rt.jar:na]
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:220) [junit-rt.jar:na]
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:53) [junit-rt.jar:na]
```

Any idea @philwebb?

**Comment by philwebb** [comment]
That's not good. The stacktrace doesn't seem to [line up with the latest code](https://github.com/spring-projects/spring-boot/blob/61502bde9a0c7534cb8e0d16de97a2f2f6d3a00d/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/config/ConfigDataEnvironmentContributor.java#L201). The version also looks a bit sus, should this be using the 2.4.3-SNAPSHOT?

**Comment by spencergibb** [comment]
No, using 2.4.2, so this will need to wait for 2.4.3 as well?

**Comment by philwebb** [comment]
Oh no, that should work. I wonder why the line numbers don't match. I can't tell what's actually `null`. Can I run that test myself?

**Comment by spencergibb** [comment]
it's actually 2.4.2-SNAPSHOT, maybe that's why line numbers are off? You can run that test, but you'd have to change ConfigServerConfigDataLoader to add the option.

**Comment by philwebb** [comment]
https://github.com/spring-projects/spring-boot/issues/25029

**Comment by spencergibb** [comment]
Closed via 587cc97b4cd3a0968fc16eaac902015e2a5b91c3
